pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -37,6 +37,7 @@ type awsCloudwatchMetadata struct {
 	TargetMetricValue           float64 `keda:"name=targetMetricValue, order=triggerMetadata"`
 	ActivationTargetMetricValue float64 `keda:"name=activationTargetMetricValue, order=triggerMetadata, optional"`
 	MinMetricValue              float64 `keda:"name=minMetricValue, order=triggerMetadata"`
+	IgnoreNullValues            bool    `keda:"name=ignoreNullValues, order=triggerMetadata, optional, default=true"`
```

Comment:
We have to maintain current behavior as there are users that can expect the error instead of the default value. Setting true as default will change how KEDA handles the null values as default
```suggestion
	IgnoreNullValues            bool    `keda:"name=ignoreNullValues, order=triggerMetadata, optional, default=false"`
```

Reply:
sorry @robpickerill, I didn0t notice this before

Reply:
Hmm its been a while but as I recall this: Currently Keda Cloudwatch scaler ignores null values, doesn't it? If this is set to false the scaler will throw an error when it gets null values, which is new behaviour.

I added this test, as a way to validate the backwards compatibility: https://github.com/kedacore/keda/blob/d55be16ecc65154cf63bdc2cda0a12cd6233cc67/tests/scalers/aws/aws_cloudwatch_min_metric_value/aws_cloudwatch_min_metric_value_test.go

Let me double check, but I think this is the correct value for ignore null values?

Reply:
Just writing this out, for myself to sanity check, and step through this to see if you come to the same conclusion:

The test here tests: ignoreNullValues: false, expecting an error and no further scaling actions, so deployment remains at 0.

https://github.com/kedacore/keda/blob/d55be16ecc65154cf63bdc2cda0a12cd6233cc67/tests/scalers/aws/aws_cloudwatch_ignore_null_values_false/aws_cloudwatch_ignore_null_values_false_test.go#L119

The test here tests the omission of ignoreNullValues, therefore it's the default value, and expects the minMetricValue to scale the workload, which is the current behaviour:

https://github.com/kedacore/keda/blob/d55be16ecc65154cf63bdc2cda0a12cd6233cc67/tests/scalers/aws/aws_cloudwatch_min_metric_value/aws_cloudwatch_min_metric_value_test.go 

Let me know your thoughts here, as it has been some time since I last looked at this

Reply:
btw, happy to accept the change proposed, just want to be sure we both on the same lines on this before I do

Reply:
uff.. you are totally right. my bad, sorry for the confusion 😓

Reply:
and thanks for not just accepting the suggestion ❤️